src/core/contourFeature.js

```diff
@@ -194,9 +198,55 @@ geo.contourFeature = function (arg) {
     if (gridW * gridH > data.length) {
       gridH = Math.floor(data.length) / gridW;
     }
+    /* If we are not using the position values (we are using x0, y0, dx, dy),
+     * and wrapLongitude is turned on, and the position spans 180 degrees,
+     * duplicate one or two columns of points at opposite ends of the map. */
+    usePos = (x0 === null || x0 === undefined || y0 === null ||
+        y0 === undefined || !dx || !dy);
+    if (!usePos && result.wrapLongitude && (x0 < -180 || x0 > 180 ||
+        x0 + dx * (gridW - 1) < -180 || x0 + dx * (gridW - 1) > 180) &&
```

Comment:
In the general case, this is pretty complicated.  For instance, points are easy, but what should lines do?  You don't necessarily want to change from drawing a line from 179E, 0N to 181E, 0N to drawing to 179E, 0N to 179W, 0N, as suddenly you've draw a line that is 358 degrees long rather than 2, and that could be very misleading.  Similarly, a polygon representing an island at straddling 180 degrees would become the width of the world.

Rather, I think the more correct action would be to *repeat* features every 360 degrees for all positions within the viewport.

Contours have the special case that they can wrap the globe, and therefore in certain instances, the grid of points should be 'closed'.  @kotfic has a case where he has a grid that is 360 data points wide with a 1 degree spacing.  We don't want to have a gap of 1 degree in the contour.  I'm not sure how that logic would extend to anything else.

So, yes, we should have a general way of repeating features so that we can zoom out, but I think it really should be instantiated in a different manner (as a completely different issue and PR).  When we do that, we can revisit contours and stop wrapping them (but still extend one extra column for a world-wide data set).